packages/eslint-plugin/src/rules/prefer-optional-chain-utils/analyzeChain.ts

```diff
@@ -242,13 +246,17 @@ function getReportRange(
 }
 
 function getReportDescriptor(
-  sourceCode: SourceCode,
+  context: RuleContext<
+    PreferOptionalChainMessageIds,
+    [PreferOptionalChainOptions]
+  >,
```

Comment:
[Question] Why the change? Unrelated/unnecessary?

Unless there's a need to switch this, could you revert the change please?

Reply:
Yeah, mistake, the change was from an older approach I tried, I'll revert